### DISCUSSION TITLE
import and using syntax

### DISCUSSION THREAD

**Discussion by RupertAvery** [Maximilian-Winter/Bite-Programming-Language#44] [General]
Most of the time, the user will want to use imported module's classes, functions directly (not sure about variables)

I still think that import/using introduces redundant code.

Is it possible to allow import to also do the same function as using all in one statement?

```
import System;

PrintLine(...);
```

And still allow specifying the module, in the event of ambiguity:

```
import System;
import MySystem;  // has custom PrintLine

PrintLine(...);  // error, ambiguous 
System.PrintLine(...);  // specific
MySystem.PrintLine(...); // specific
```

## Current behavior

Right now, there is no warning for ambiguity, and importing /using two modules with the same methods will end up with the FIRST imported module's declarations winning.

```
module DepA

function foo() {
    return "foo";
}
```

```
module DepB

function foo() {
    return "bar";
}
```

```
module Main

import DepA;
using DepA;
import DepB;
using DepB;

var a  = foo();  // returns "foo"
```

**Comment by vorov2** [comment]
Import + using seem really redundant. I would expect an import statement to do all I need. In the case of ambiguity there are options:

1. Explicit import of names (no namespace pollution with bindings you never going to need):

```
import System (Print, PrintLine);
import OtherModule (PrintLine); //error, PrintLine already declarated
```

2. Shadowing with implicit imports:

```
import System;
import OtherModule;

PrintLine(...) //using implementation from OtherModule
System.PrintLine(...) //Explicit
```

3. Error on any ambiguity 

Some languages combine both 1 and 2 using different statements, e.g. import (explicit list needed), open (everything is imported implicitly)
